resources/arduino_files/camera_board/camera_board.ino

```diff
@@ -5,162 +5,261 @@
+
+void main_loop() {
+  unsigned long now = millis();
+  if (next_report_millis <= now) {
+    // Schedule the next report for `interval' milliseconds from the last report,
+    // unless we've fallen behind.
+    constexpr int interval = 1000;
+    next_report_millis += interval;
+    if (next_report_millis <= now) {
+      next_report_millis = now + interval;
+    }
+
+    // Collect the data. Since some of these operations take a while, keep updating the
+    // LED as appropriate. Could probably be done with an interrupt handler instead.
+    report_num++;
+    acc_handler.collect();
+    led_handler.update();
+    dht_handler.collect();
+    led_handler.update();
+    bool cam0 = digitalRead(CAM_0_RELAY);
+    bool cam1 = digitalRead(CAM_1_RELAY);
+
+    // Format/output the results.
+    Serial.print("{\"name\":\"camera_board\", \"count\":");
+    led_handler.update();
+    Serial.print(millis() - end_setup_millis);
+    led_handler.update();
+    Serial.print(", \"num\":");
+    led_handler.update();
+    Serial.print(report_num);
+    led_handler.update();
+    Serial.print(", \"inputs\":");
+    led_handler.update();
+    Serial.print(inputs);
+    led_handler.update();
+    Serial.print(", \"camera_00\":");
+    led_handler.update();
+    Serial.print(cam0);
+    led_handler.update();
+    Serial.print(", \"camera_01\":");
+    led_handler.update();
+    Serial.print(cam1);
+    led_handler.update();
+    acc_handler.report();
+    led_handler.update();
+    dht_handler.report();
+    led_handler.update();
+    Serial.println("}");
+    led_handler.update();
+    Serial.flush();
+    led_handler.update();
+  }
+
   // Read any serial input
-  //    - Input will be two comma separated integers, the
+  //    - Input will be two integers (with anything in between them), the
   //      first specifying the pin and the second the status
   //      to change to (1/0). Cameras and debug led are
   //      supported.
   //      Example serial input:
-  //           4,1   # Turn fan on
+  //           5,1   # Turn camera 0 on
+  //           6,0   # Turn camera 1 off
   //          13,0   # Turn led off
   while (Serial.available() > 0) {
+    inputs++;
     int pin_num = Serial.parseInt();
     int pin_status = Serial.parseInt();
 
     switch (pin_num) {
-    case CAM_01_RELAY:
-    case CAM_02_RELAY:
-      if (pin_status == 1) {
-        turn_pin_on(pin_num);
-      } else if (pin_status == 0) {
-        turn_pin_off(pin_num);
-      } else if (pin_status == 9) {
-        toggle_pin(pin_num);
-      }
-      break;
-    case RESET_PIN:
-      if (pin_status == 1) {
-        turn_pin_off(RESET_PIN);
-      }
-      break;
-    case LED_BUILTIN:
-      digitalWrite(pin_num, pin_status);
-      break;
+      case CAM_0_RELAY:
+      case CAM_1_RELAY:
+        if (pin_status == 1) {
+          turn_pin_on(pin_num);
+        } else if (pin_status == 0) {
+          turn_pin_off(pin_num);
+        }
+        break;
+      case LED_BUILTIN:
```

Comment:
We might be able to remove this since the led_handler will just clobber it anyway.

Reply:
OK